### PR TITLE
New tests added

### PR DESCRIPTION
**Pull request by ClaudiaLSS**

### CODE REVIEW

**Comment by Bachibouzouk** [comment]
I see that the branch `tests` on which this PR is directed is no different than main branch, is that normal @mohammadamint ?

File: tests/test_peak_time_range.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    peak_enlarge, user_list, num_profiles = initialise_inputs(
+        j, fname, num_profiles=1
+    )
+    return user_list
+
+class TestPeakTime:
+    def test_calc_peak_time_range(self):
+            j=1
+            peak_enlarge=0.15
+            user_list=load_usecase(j,peak_enlarge )
```
Comment (Bachibouzouk) [inline tests/test_peak_time_range.py:31]:
```suggestion
            user_list=load_usecase(j)
```
Here `load_usecase` should only take one argument

File: tests/test_calc_peak_time_range.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        peak_time_range = calc_peak_time_range(user_list, peak_enlarge=0)
+        assert isinstance(peak_time_range, np.ndarray)
+        assert peak_time_range.shape[0] > 0
+        assert peak_time_range.dtype == int
+        assert np.all(peak_time_range >= 0)
+
+    def test_calc_peak_time_range_with_inputs(self):
+            j=1
+            peak_enlarge=0.15
+            user_list=load_usecase(j,peak_enlarge )
```
Comment (Bachibouzouk) [inline tests/test_calc_peak_time_range.py:84]:
```suggestion
            user_list=load_usecase(j)
```

File: tests/test_peak_time_range.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import random
+import math
+from scipy.stats import norm
+
+from ramp.core.core import User, Appliance
+from ramp.core.core import UseCase
+from ramp.core.initialise import initialise_inputs
+from ramp.core.stochastic_process import calc_peak_time_range
+
+def load_usecase(j=None, fname=None):
```
Comment (Bachibouzouk) [inline tests/test_peak_time_range.py:21]: If you reuse the same function several times then define it under `tests/utils.py` and import it. That way it is easier to maintain the tests on the long term

File: tests/test_peak_time_range.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import pytest
+import numpy as np
+import random
+import math
+from scipy.stats import norm
+
+from ramp.core.core import User, Appliance
+from ramp.core.core import UseCase
+from ramp.core.initialise import initialise_inputs
+from ramp.core.stochastic_process import calc_peak_time_range
```
Comment (Bachibouzouk) [inline tests/test_peak_time_range.py:19]:
```suggestion
import numpy as np

from ramp.core.initialise import initialise_inputs
from ramp.core.stochastic_process import calc_peak_time_range
```
In the suggestion I removed the unused imports

File: tests/test_calc_peak_time_range.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    )
+    return user_list
+
+class TestCalcPeakTimeRange:
+
+    # Tests that the peak time range is calculated correctly for a single user class with the default peak_enlarge value
+    def test_single_user_class_default_peak_enlarge(self):
+        user_list = [User(maximum_profile=np.array([1, 2, 3, 4, 5]))]
+        peak_time_range = calc_peak_time_range(user_list)
+        assert isinstance(peak_time_range, np.ndarray)
```
Comment (Bachibouzouk) [inline tests/test_calc_peak_time_range.py:28]: As the return value of `calc_peak_time_range` is the return of `np.arange()` function I don't think it makes sense to test for np.ndarray here

File: tests/test_calc_peak_time_range.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        assert np.all(peak_time_range >= 0)
+
+    # Tests that the peak time range is calculated correctly for a single user class with a non-default peak_enlarge value
+    def test_single_user_class_non_default_peak_enlarge(self):
+        user_list = [User(maximum_profile=np.array([1, 2, 3, 4, 5]))]
+        peak_time_range = calc_peak_time_range(user_list, peak_enlarge=0.2)
+        assert isinstance(peak_time_range, np.ndarray)
+        assert peak_time_range.shape[0] > 0
+        assert peak_time_range.dtype == int
+        assert np.all(peak_time_range >= 0)
```
Comment (Bachibouzouk) [inline tests/test_calc_peak_time_range.py:50]: Those are the same `assert` than for all the other tests, shouldn't be the test more specific to the change of `peak_enlarge` value in this case?

File: tests/test_peak_time_range.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python3
```
Comment (Bachibouzouk) [inline tests/test_peak_time_range.py:1]: How is this test different than `test_calc_peak_time_range.py::test_calc_peak_time_range_with_inputs` ?

File: tests/test_rand_total_time_of_use.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    
+    #Test that the method returns an integer
+    @pytest.mark.usefixtures("appliance_instance")        
+    def test_returns_integer_value(self, appliance_instance):
+        result = appliance_instance.rand_total_time_of_use([0, 480], [600, 1080], [1200, 1440])
+        assert isinstance(result, int)
+    
+    #Tests that the method returns a value greater than func_cycle
+    @pytest.mark.usefixtures("appliance_instance")    
+    def test_rand_time_equal_or_greater_than_func_cycle(self, appliance_instance):
```
Comment (Bachibouzouk) [inline tests/test_rand_total_time_of_use.py:43]: I think for such random function the test should run on at least 100  iteration of the method and make sure for none of these call is the `rand_time < func_cycle`

**Review by Bachibouzouk — COMMENTED** [review]
Thanks for your efforts @ClaudiaLSS it feels good to have some tests !

I managed to make all test pass (except the tests where `maximum_profile` is used in the `User` class).

I made suggestion that you can accept in batch and group into one single commit (from github directly) or in several commits as you wish.

A general comment is to delete the unused import statements (I see them greyed out in pycharm)

I would like also that we start linting the code using `black` it is a pip package you can install and simply run `black tests/` and it lints automatically your files

I would suggest that changes are first made such as to be able to run the tests and that they pass, then we merge and you can improve the tests in a subsequent PR

File: tests/test_peak_time_range.py (hunk 7 above)
Comment (ClaudiaLSS) [reply tests/test_peak_time_range.py:1]: Thanks a lot for the review! I think the file [test_peak_time_range.py] was not supposed to be there, the real one was calc_peak_time_range.py. The first one needs to be deleted, sorry!

Comment (Bachibouzouk) [reply tests/test_peak_time_range.py:1]: No problems, this is what reviews are for :)

**Comment by Bachibouzouk** [comment]
The failing doc is being tackled in #85 and is due to deprecated config files, this should not impact this PR :)

**Comment by FLomb** [comment]
I have just approved #85; should we rebase the `RAMP-project:tests` branch to be up-to-date with the fix? And check if that solves the problem with the failing doc? @Bachibouzouk you are the one who can do this best :)

**Comment by Bachibouzouk** [comment]
> I have just approved #85; should we rebase the `RAMP-project:tests` branch to be up-to-date with the fix? And check if that solves the problem with the failing doc? @Bachibouzouk you are the one who can do this best :)

Maybe Claudia can simply merge development into her branch and we merge dev into `RAMP-project:tests` because it is easier.  Then we would rebase `RAMP-project:tests` on `development` before merging it into `development`

**Comment by Bachibouzouk** [comment]
@ClaudiaLSS - what is your stand on this PR?

**Comment by ClaudiaLSS** [comment]
> Maybe @ClaudiaLSS can simply merge development into her branch and we merge dev into `RAMP-project:tests` because it is easier. Then we would rebase `RAMP-project:tests` on `development` before merging it into `development`

Hi @Bachibouzouk ! I'll try this.

**Comment by ClaudiaLSS** [comment]
Hi @Bachibouzouk! I think now dev and test branches are identical. I still don't have much experience with this but can you see if everything is ok?
Thanks!

**Comment by Bachibouzouk** [comment]
@ClaudiaLSS - you merged `main` instead of `developement`, I think the easier is that I will revert your last commit and then we merge your branch into `test`

**Comment by Bachibouzouk** [comment]
This PR is continued in #93

**Pull request closed without merging**